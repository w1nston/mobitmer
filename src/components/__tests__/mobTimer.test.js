import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import MobTimer from '../mobTimer';

describe('<MobTimer/>', () => {
  test('renders', () => {
    const component = shallow(<MobTimer />);
    expect(toJson(component)).toMatchSnapshot();
  });

  test('onClick triggers prop onStart', () => {
    const onStart = jest.fn();
    const component = shallow(<MobTimer onStart={onStart} />);
    component.find('button').simulate('click');
    expect(onStart).toHaveBeenCalled();
  });
});
