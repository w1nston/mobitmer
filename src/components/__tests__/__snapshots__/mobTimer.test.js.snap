exports[`<MobTimer/> renders 1`] = `
<section>
  <button>
    Start
  </button>
</section>
`;
